test/e2e/util_test.go

```diff
@@ -369,76 +337,169 @@ func fetchCatalogSourceOnStatus(crc versioned.Interface, name, namespace string,
 	return fetched, err
 }
 
-func createFieldNotEqualSelector(field string, names ...string) string {
+// createFieldNotEqualSelector generates a field selector that matches resources that have a field value that DOES NOT match any of a set of values.
+// This function panics if the generated selector cannot be parsed.
+func createFieldNotEqualSelector(field string, values ...string) fields.Selector {
 	var builder strings.Builder
-	for i, name := range names {
+	for i, value := range values {
 		builder.WriteString(field)
 		builder.WriteString("!=")
-		builder.WriteString(name)
-		if i < len(names)-1 {
+		builder.WriteString(value)
+		if i < len(values)-1 {
 			builder.WriteString(",")
 		}
 	}
 
-	return builder.String()
+	selector, err := fields.ParseSelector(builder.String())
+	if err != nil {
+		panic(fmt.Errorf("failed to build fields-not-equal selector: %s", err))
+	}
+
+	return selector
 }
 
-func cleanupOLM(namespace string) {
-	var immediate int64 = 0
-	crc := newCRClient()
-	c := newKubeClient()
+// const immediately int64 = 0
+
+var (
+	persistentCatalogNames        = []string{olmConfigMap}
+	ephemeralCatalogFieldSelector = controllerclient.MatchingFieldsSelector{
+		Selector: createFieldNotEqualSelector("metadata.name", persistentCatalogNames...),
+	}
+	persistentConfigMapNames         = []string{olmConfigMap}
+	ephemeralConfigMapsFieldSelector = controllerclient.MatchingFieldsSelector{
+		Selector: createFieldNotEqualSelector("metadata.name", persistentConfigMapNames...),
+	}
+	persistentCSVNames        = []string{packageServerCSV}
+	ephemeralCSVFieldSelector = controllerclient.MatchingFieldsSelector{
+		Selector: createFieldNotEqualSelector("metadata.name", persistentCSVNames...),
+	}
+)
+
+// TearDown deletes all OLM resources in the corresponding namespace and at the cluster scope.
+func TearDown(namespace string) {
```

Comment:
it would be useful if there were a way to control this (i.e. via an env var) so that locally I can quickly switch cleanup off to look at the cluster state.

And I think really what we want is a sort of "must-gather" per test. Doesn't need to block this PR but I'd be interesting in something that say, read all of this content off the cluster and saved it to configmaps before deleting everything.

Reply:
@benluddy and I were just talking about this over Slack.

He recommended implementing a [Reporter](https://pkg.go.dev/github.com/onsi/ginkgo/reporters?tab=doc#Reporter) to do this. We were thinking that for an initial pass we could export all resources in the `operators.coreos.com` API group to a tar file hosted in cloud storage at the end of each failed spec, then the reporter could post a link to each export in the suite summary.

Reply:
That sounds good, but it would be nice if we can make it work across github actions, openshift ci, and locally.

Reply:
> That sounds good, but it would be nice if we can make it work across github actions, openshift ci, and locally.

It may be good enough to accept a template string via environment variables to construct those URLs and set it appropriately in the corresponding config.